.github/workflows/secrets-detection.yaml

```diff
@@ -0,0 +1,76 @@
+name: Secret Detection Workflow
+on:
+    push:
+        branches:
+            - main
+    pull_request:
+        branches:
+            - main
+
+jobs:
+    secret-detection:
+        runs-on: ubuntu-latest
+        steps:
+            -
+                name: Checkout code
+                uses: actions/checkout@v2
```

Comment:
@nutjob4life per [the warning here](https://github.com/NASA-PDS/template-repo-python/actions/runs/6856766971), can we upgrade checkout to the latest version? Otherwise, once that is fixed, good to rollout to all repos. (May need to make same upgrade on Java template)

Reply:
I just fixed it up

Reply:
@jordanpadams thanks for the bump! 🎉